### PR TITLE
feat: add password settings columns

### PR DESCRIPTION
**Pull request by kimskimchi**
## Summary

Introduces password settings as columns in database. 

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Updated associated configuration where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

## Related Issues

Resolves #

### CODE REVIEW

File: internal/server/models/settings.go [hunk 1]
```diff
@@ -5,4 +5,10 @@ type Settings struct {
 
 	PrivateJWK EncryptedAtRestBytes
 	PublicJWK  []byte
+
+	LowercaseMin int `gorm:"default:0"`
+	UppercaseMin int `gorm:"default:0"`
+	NumberMin    int `gorm:"default:0"`
+	SymbolMin    int `gorm:"default:1"`
```
[the remark this answers is not in the document]
Comment (kimskimchi) [reply internal/server/models/settings.go:12]: It seems to be the usual the default for most passwords (https://en.wikipedia.org/wiki/Password_policy), moreso than enforcing at least one number/upper.

File: internal/server/data/migrationdata/202207120000-postgres.sql [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+    created_at timestamp with time zone,
+    updated_at timestamp with time zone,
+    deleted_at timestamp with time zone,
+    private_jwk bytea,
+    testing_jwk bytea,
+    lowercase_min bigint DEFAULT 0,
+    uppercase_min bigint DEFAULT 0,
+    number_min bigint DEFAULT 0,
+    symbol_min bigint DEFAULT 1,
+    length_min bigint DEFAULT 8
```
Comment (BruceMacD) [inline internal/server/data/migrationdata/202207120000-postgres.sql:427]: I think you want these columns to not be in the table in order to test the migration

File: internal/server/data/migrationdata/202207120000-postgres.sql [hunk 3]
```diff
@@ -0,0 +1,887 @@
...
...22-07-13 14:57:13.369333+00', '2022-07-13 14:57:13.369333+00', NULL, '\x41414141346c526b61426847634e556c4739537278626e327036526a506a75714578522b7064325872722f472b384b376b52545545767844765a4f347a4759506d7a736a502b4139707857546664544e45512b7a6d69324d6836324352425830644351737a3833744b50753455596b4b434871714741332b4b576447754444656b6a7147705a6c67544d57396232794a6c544d7831377862726850394f2f66456770306b5052717954717a766264556d7464415168695a7962475531456642434b666b6a59686d412f446b79667838555376514a6942554b4f39666945626e4435744b7539385a5559725957326478387567326669435346704c614f424343613865554b6f45304f6b2b54416964584d4b4278487135535866375870787a54776f5032315837697a5a41364a35774b56524f77475957567a5a324e74424276506947596d4c335a6863693973615749766157356d636d466f6353397a5a584a325a584976633346736158526c4d79356b596935725a586b4d42537a4973784d73546d4f747859522b', '\x7b22757365223a22736967222c226b7479223a224f4b50222c226b6964223a22463374704a7441677a5839375978373831335a4e593244516c574177533958696264777034345a754a30303d222c22637276223a2245643235353139222c22616c67223a2245443235353139222c2278223a2273777161694e697a41313677544957796b446f34636678576875746170796f336d303370376e4436446c38227d', 0, 0, 0, 1, 8);
```
Comment (BruceMacD) [inline internal/server/data/migrationdata/202207120000-postgres.sql:619]:
```suggestion
INSERT INTO testing.settings (id, created_at, updated_at, deleted_at, private_jwk, testing_jwk, lowercase_min, uppercase_min, number_min, symbol_min, length_min) VALUES (70166651577114624, '2022-07-13 14:57:13.369333+00', '2022-07-13 14:57:13.369333+00', NULL, '\x41414141346c526b61426847634e556c4739537278626e327036526a506a75714578522b7064325872722f472b384b376b52545545767844765a4f347a4759506d7a736a502b4139707857546664544e45512b7a6d69324d6836324352425830644351737a3833744b50753455596b4b434871714741332b4b576447754444656b6a7147705a6c67544d57396232794a6c544d7831377862726850394f2f66456770306b5052717954717a766264556d7464415168695a7962475531456642434b666b6a59686d412f446b79667838555376514a6942554b4f39666945626e4435744b7539385a5559725957326478387567326669435346704c614f424343613865554b6f45304f6b2b54416964584d4b4278487135535866375870787a54776f5032315837697a5a41364a35774b56524f77475957567a5a324e74424276506947596d4c335a6863693973615749766157356d636d466f6353397a5a584a325a584976633346736158526c4d79356b596935725a586b4d42537a4973784d73546d4f747859522b', '\x7b22757365223a22736967222c226b7479223a224f4b50222c226b6964223a22463374704a7441677a5839375978373831335a4e593244516c574177533958696264777034345a754a30303d222c22637276223a2245643235353139222c22616c67223a2245443235353139222c2278223a2273777161694e697a41313677544957796b446f34636678576875746170796f336d303370376e4436446c38227d');
```

To test the migration

**Review by BruceMacD — COMMENTED** [review]
Almost there, just needs a tweak to the Postgres migration test data

File: internal/server/models/settings.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kimskimchi) [reply internal/server/models/settings.go]: reset to 0

**Review by BruceMacD — APPROVED** [review]

**Pull request merged**